lib/vrl/core/src/prelude.rs

```diff
@@ -24,7 +24,9 @@ pub use std::fmt;
 
 // commonly used function types
 
-pub use compiler::function::{ArgumentList, Compiled, Example, Parameter};
+pub use compiler::function::{
+    ArgumentList, CompileInfo as FunctionCompileInfo, Compiled, Example, Parameter,
```

Comment:
I'm not personally keen on this as it makes grepping harder. For simplicity, it would be better to just call it `FunctionCompileInfo` throughout. 

Or, what about `CompileContext`? We have a `Context` passed in for the runtime, so name would mirror that.

Reply:
👍 I just figured `compiler::function::FunctionCompileInfo` was a bit redundant in naming. I don't have strong opinions though, I'll update.

I considered `CompileContext` but this struct seems specific to function compilation, for now at least.